lib/new_relic/agent/instrumentation/view_component/instrumentation.rb

```diff
@@ -0,0 +1,40 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic::Agent::Instrumentation
+  module ViewComponent
+    INSTRUMENTATION_NAME = NewRelic::Agent.base_name(name)
+
+    def render_in_with_tracing(*args)
+      NewRelic::Agent.record_instrumentation_invocation(INSTRUMENTATION_NAME)
+
+      component = self.class.name
+      identifier = self.class.identifier
+
+      begin
+        segment = NewRelic::Agent::Tracer.start_segment(
+          name: metric_name(identifier, component)
+        )
```

Comment:
To reduce object allocations at the cost of making things less readable, you could remove this variable assignment and access the component and identifier names in the method arguments: 
```suggestion
      begin
        segment = NewRelic::Agent::Tracer.start_segment(
          name: metric_name(self.class.name, self.class.identifier)
        )
```